backend/controllers/vote.js

```diff
@@ -63,6 +63,7 @@ exports.vote = async (req, res) => {
       route.votes.push({"username": username, "score": score})
     }
 
+    route.markModified('votes')
     await route.save();
```

Comment:
If we try to update an existing field in an array in MongoDB, the array needs to be marked as modified before calling `save`. Otherwise, the change will not be picked up.